src/genymotion.js

```diff
@@ -0,0 +1,224 @@
+import { expandPath } from 'appcd-path';
+import { isDir, isFile } from 'appcd-fs';
+
+/**
+ * Common Genymotion install locations
+ * @type {Object}
+ */
+export const genymotionLocations = {
+	darwin: [
+		'/Applications/Genymotion.app/',
+		'~/Applications/Genymotion.app/'
+	],
+	linux: [
+		'/opt',
+		'/usr',
+		'~'
+	],
+	win32: [
+		'%ProgramFiles%\\Genymobile\\Genymotion',
+		'%ProgramFiles%\\Genymotion',
+		'%ProgramFiles(x86)%\\Genymobile\\Genymotion',
+		'%ProgramFiles(x86)%\\Genymotion'
+	]
+};
+
+/**
+ * Common Genymotion home directory locations
+ * @type {Object}
+ */
+export const genymotionHomeLocations = {
+	darwin: [
+		'~/.Genymobile/Genymotion',
+		'~/.Genymotion'
+	],
+	linux : [
+		'~/.Genymobile/Genymotion',
+		'~/.Genymotion'
+	],
+	win32: [
+		'%LocalAppData%/Genymobile/Genymotion'
+	]
+};
+
+export const genymotionPlist = '~/Library/Preferences/com.genymobile.Genymotion.plist';
+
+/**
+ * Genymotion information
+ */
+export class Genymotion {
+	/**
+	 * Performs tests to see if this is a Genymotion install directory,
+	 * and then initializes the info.
+	 *
+	 * @param {String} dir - Directory to scan.
+	 * @access public
+	 */
+	constructor(dir) {
+		if (typeof dir !== 'string' || !dir) {
+			throw new TypeError('Expected directory to be a valid string');
+		}
+
+		dir = expandPath(dir);
+		if (!isDir(dir)) {
+			throw new Error('Directory does not exist');
+		}
+
+		// on OS X, it lives in Contents/MacOS
+		if (process.platform === 'darwin') {
+			let p = path.join(dir, 'Contents', 'MacOS');
+			if (isDir(p)) {
+				dir = p;
+			} else {
+				p = path.join(dir, '..', 'Contents', 'MacOS');
+				if (isDir(p)) {
+					dir = p;
+				}
+			}
+		}
+
+		this.emulators 	 = [];
+		this.executables = {};
+		this.home 		 = null;
+		this.path 		 = dir;
+		if (process.platform === 'darwin') {
+			this.deployedDir = expandPath(plist.readFileSync(expandPath(genymotionPlist))['vms·path']);
+		} else {
+			this.deployedDir = null;
+		}
+		this.executables.genymotion = path.join(dir, `genymotion${exe}`);
+
+		if (process.platform === 'darwin') {
+			this.executables.player = path.join(dir, 'player.app', 'Contents', 'MacOS', 'player');
+		} else {
+			this.executables.player = path.join(dir, `player${exe}`);
+		}
+
+		for (const name of Object.keys(this.executables)) {
+			if (!isFile(this.executables[name])) {
+				throw new Error(`Directory does not contain the "${name}" executable`);
+			}
+		}
+
+		const homeDirs = genymotionHomeLocations[process.platform];
+		for (let homeDir of homeDirs) {
+			homeDir = expandPath(homeDir);
+			if (isDir(homeDir)) {
+				this.home = homeDir;
+				break;
+			}
+		}
+	}
+
+	async init(vbox) {
+		const detectRegistry = async () => {
+			if (process.platform === 'win32') {
+				let deployedDir;
+				try {
+					deployedDir = await registry.get('HKCU', '\\Software\\Genymobile\\Genymotion', 'vms.path');
+				} catch (ex) {
+					// squelch
+				}
+
+				return expandPath(deployedDir);
```

Comment:
Don't `expandPath()` if `deployedDir` is `undefined` due to an exception getting the value from the registry.

Reply:
Also, just set `this.deployedDir`.